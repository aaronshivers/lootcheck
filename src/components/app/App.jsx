import React from 'react';
import WalletComponent from '../wallet/Wallet';
import LootComponent from '../loot/Loot';

const App = () => (
  <div>
    <h1>Loot Check</h1>
    <hr />
    <WalletComponent />
    <hr />
    <LootComponent />
    <a
      href="https://www.coindesk.com/price/bitcoin"
      target="_blank"
      rel="noreferrer"
    >
      Powered by CoinDesk
    </a>
  </div>
);

App.propTypes = {};

export default App;
