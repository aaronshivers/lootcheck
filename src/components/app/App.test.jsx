import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const app = shallow(<App />);

  it('should render correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('should contain a connected Wallet component', () => {
    expect(app.exists('Connect(Wallet)')).toBeTruthy();
  });

  it('should contain a connected Loot component', () => {
    expect(app.exists('Connect(Loot)')).toBeTruthy();
  });

  it('should contain a link to the coindesk price page', () => {
    const message = 'Powered by CoinDesk';
    const url = 'https://www.coindesk.com/price/bitcoin';
    expect(app.find('a').text()).toBe(message);
    expect(app.find('a').props().href).toBe(url);
  });
});
