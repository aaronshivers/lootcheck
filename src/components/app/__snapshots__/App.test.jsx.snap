// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <h1>
    Loot Check
  </h1>
  <hr />
  <Connect(Wallet) />
  <hr />
  <Connect(Loot) />
  <a
    href="https://www.coindesk.com/price/bitcoin"
    rel="noreferrer"
    target="_blank"
  >
    Powered by CoinDesk
  </a>
</div>
`;
